plugins/single_plugins/wsets.cpp

```diff
@@ -113,6 +121,48 @@ class wayfire_wsets_plugin_t : public wf::plugin_interface_t
     std::list<wf::activator_callback> send_callback;
     std::map<int, std::shared_ptr<wf::workspace_set_t>> available_sets;
 
+    wf::ipc::method_callback attach_to_output = [=] (nlohmann::json data)
+    {
+        WFJSON_EXPECT_FIELD(data, "id", number_integer);
+        WFJSON_EXPECT_FIELD(data, "output", number_integer);
+        auto wset = wf::ipc::find_workspace_set_by_id(data["id"]);
+        if (!wset)
+        {
+            return wf::ipc::json_error("workspace set not found");
+        }
+
+        auto o = wf::ipc::find_output_by_id(data["output"]);
+        if (!o)
+        {
+            return wf::ipc::json_error("output not found");
+        }
+
+        auto response = wf::ipc::json_ok();
+        wset->attach_to_output(o);
```

Comment:
This has the potential to crash Wayfire, if we move one output's active workspace set to another output: https://github.com/WayfireWM/wayfire/blob/master/src/output/workspace-impl.cpp#L336

I am in general wondering about the usefulness of this particular IPC call. If we do not immediately make the wset visible on another output, what is the use of this operation? Do you have a particular use-case?

Reply:
There is no need to pre-create workspace sets, the wsets plugin automatically creates a wset when it is needed. Otherwise you're wasting resources :)